From cc9aae3d2d40432c03b235656da84949184af511 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 21 Nov 2018 16:53:31 +0300
Subject: [PATCH] luci-mod-system: output only existing timezones

Check zone existence in /usr/share/zoneinfo. Output to combobox
only existing timezones.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 modules/luci-mod-system/luasrc/model/cbi/admin_system/system.lua | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modules/luci-mod-system/luasrc/model/cbi/admin_system/system.lua b/modules/luci-mod-system/luasrc/model/cbi/admin_system/system.lua
index c26fd475a..fb040f0f3 100644
--- a/modules/luci-mod-system/luasrc/model/cbi/admin_system/system.lua
+++ b/modules/luci-mod-system/luasrc/model/cbi/admin_system/system.lua
@@ -45,7 +45,10 @@ o = s:taboption("general", ListValue, "zonename", translate("Timezone"))
 o:value("UTC")
 
 for i, zone in ipairs(zones.TZ) do
-	o:value(zone[1])
+	local zone_exists = fs.access("/usr/share/zoneinfo/" .. zone[1])
+	if (zone_exists) then
+		o:value(zone[1])
+	end
 end
 
 function o.write(self, section, value)
-- 
2.16.2.windows.1

